File: Data Structures & Algorithms/jump-game/submission-5.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        int goal = n-1;
        int index = n-2;

        while(index >= 0)
        {
            if(index + nums[index] >= goal)
            {
                goal = index;   
            }
            index--;
        }

        return goal == 0 ? true : false;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-5.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        visit = set()

        for n in nums:
            if n in visit:
                return True;
            visit.add(n)

        return False

        # ordereddict
        # dequeue - stack and queue
        # list, set, dictionary, defaultdict(main)
        # heap - list (heapify), heap operations,min(default), max --
        # stack - list or dequeue
        # queue  - dequeue
        # string immutable slicing
        # array not existing
        # list 
        # dont use loop when appending 2 lists, use extend

         